lib/puppet.rb

```diff
@@ -9,7 +9,7 @@
 # see the bottom of the file for further inclusions
 # Also see the new Vendor support - towards the end
 #
-require 'facter'
+require ENV['USE_FACTER_NG'] ? 'facter-ng' : 'facter'
```

Comment:
I don't know if it is cleaner/better/ruby-style, one alternative could be to have a facter-wrapper.rb with facter selection logic (currently `ENV['USE_FACTER_NG'] ? 'facter-ng' : 'facter'`) and require this `facter-wrapper` in puppet code

Reply:
This is how we handled it previously when we switched from facter 2.x to cfacter: f41800ab4c0a928ff4f38ad5d1f6bd1e2b5619cd I think doing that is preferable as the implementation can be specified as a puppet setting, which is helpful when trying to run acceptance tests (since you only need to add the setting in a presuite step) and not every invocation of `puppet`

Reply:
@joshcooper I've updated the code today to use puppet settings, just trying to figure out why some specs fail.